### PR TITLE
[web][api]Fix/date issues students supporting users

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Created a Typeorm transformer to allow us to map a Date column on DB to a Date object on the typescript. The time will be always defined as a local time zero. Typeorm would map a date-only DB column to a string, as per the references below. 
Some references:
- https://wanago.io/2021/03/15/postgresql-typeorm-date-time/
- https://stackoverflow.com/questions/62696628/how-can-i-create-columns-with-type-date-and-type-datetime-in-nestjs-with-typeorm
- https://github.com/typeorm/typeorm/issues/2176
image
- Birthdate on the students model was renamed to be in sync with supporting_users and database. Apparently, "birthDate" and "birthdate" are both acceptable, but since the DB column is named as birth_date it seems more logical use birthDate.
- Changed Date component on the search UI on supporting users because the PrimeVue calendar was not behaving as expected and it was causing a bad user experience.

### CODE REVIEW

**Review by abschwenker — APPROVED** [review]
lgtm

**Review by dheepak-aot — APPROVED** [review]
LGTM

File: sources/packages/api/src/database/transformers/date-only.transformer.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * be mapped to a Date object with local time always
+ * defined as zero.
+ * When mapping a Date only field from database Typeorm
+ * will use a string object. This transformer will allow
+ * the conversion to a javascript Date object to avoid the
+ * issues while converting a string with no time specified,
+ * as the one returned from the database.
+ */
+export const dateOnlyTransformer: ValueTransformer = {
+  from: (dbValue: any): any => {
```
Comment (dheepak-aot) [inline sources/packages/api/src/database/transformers/date-only.transformer.ts:29]: Just checking, is it required by ValueTransformer that we should use any. If else we can use the data type(string|Date).

Comment (andrewsignori-aot) [reply sources/packages/api/src/database/transformers/date-only.transformer.ts:29]: I wouldn't say required, but I followed what was defined on ValueTransformer.

image

**Review by ann-aot — APPROVED** [review]
LGTM 👍 Nice that we found that issue

File: sources/packages/api/src/database/transformers/date-only.transformer.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/database/transformers/date-only.transformer.ts]: Changed as we talked. Thanks for the recommendation, looks better now 😉

**Comment by andrewsignori-aot** [comment]
> LGTM 👍 Nice that we found that issue

Thanks for the collaboration in helping testing and identify the gaps 😉

File: sources/packages/api/src/database/transformers/date-only.transformer.ts (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/api/src/database/transformers/date-only.transformer.ts]: Thanks for making the changes.

**Comment by dheepak-aot** [comment]
LGTM again with dateOnlyTransformer. 👍

**Review by abschwenker — APPROVED** [review]
lgtm

**Review by guru-aot — APPROVED** [review]
LGTM @andrewsignori-aot

**Pull request merged**